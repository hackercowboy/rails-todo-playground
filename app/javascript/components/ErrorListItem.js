import React from 'react';
import PropTypes from 'prop-types';
import htm from 'htm';

const html = htm.bind(React.createElement);

function ErrorListItem({ error }) {
  return html`<li className="error-list-item">${error}</li>`;
}

ErrorListItem.propTypes = {
  error: PropTypes.string.isRequired,
};

export default ErrorListItem;
