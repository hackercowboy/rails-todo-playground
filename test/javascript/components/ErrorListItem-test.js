import React from 'react';
import { render, prettyDOM } from '@testing-library/react';

import ErrorListItem from '../../../app/javascript/components/ErrorListItem';

const e = React.createElement;

describe('<ErrorListItem/>', () => {
  it('should render', () => {
    render(e(ErrorListItem, { error: 'Test Error' }));
    expect(prettyDOM()).toMatchSnapshot();
  });
});
