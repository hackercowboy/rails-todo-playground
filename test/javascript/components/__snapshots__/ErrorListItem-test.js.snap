// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorListItem/> should render 1`] = `
"[36m<body>[39m
  [36m<div>[39m
    [36m<li[39m
      [33mclass[39m=[32m\\"error-list-item\\"[39m
    [36m>[39m
      [0mTest Error[0m
    [36m</li>[39m
  [36m</div>[39m
[36m</body>[39m"
`;
